monitoring/gitlab.py

```diff
@@ -0,0 +1,140 @@
+        gitlab_template = self.get_template(context, template_path)
+        gitlab_template.globals["urlencode"] = urllib.parse.quote
+
+        server_url = "https://git.smhi.se"
+        gl = gitlab.Gitlab(server_url, private_token=key)
+
+        self.log_debug("Connecting to Gitlab.")
+
+        project, issue = self.get_project_and_issue(gl, issue_subject)
+
+        self.log_debug("Uploading attachments.")
+        issue.uploads = []
+        for item in presentation_list:
+            if item["presentation_type"] == "image":
+                file_name = Path(item["path"]).name
+                upload = project.upload(file_name, filepath=f"{dst_folder}/{file_name}")
+                item["image_link"] = upload[
+                    "markdown"
+                ]  # string of the form "![]()" generated by Gitlab
+
+        # render the template and add as description
+        self.log_debug("Updating the issue description.")
+        issue.description = gitlab_template.render(
+            presentation_list=presentation_list,
+            issue_url=issue.web_url,
+            create_anchor=create_anchor,
+        )
+        self.log_debug("Saving issue.")
+        issue.save()
+
+    def get_presentation_list(self, sources, dst_folder):
+        """create a list of presentation objects"""
+        self.log_debug("Getting list of presentation objects.")
+        presentation_list = []
+        for src in sources:
+            try:
+                try:
+                    pres_object = PresentationObject(dst_folder, **src)
+                except TypeError:
+                    pres_object = PresentationObject(dst_folder, src)
+                self.log_debug(
+                    f"Loading {pres_object.loader.diag_type} diagnostic from {pres_object.loader.path}."
+                )
+                presentation_list.append(pres_object.create_dict())
+            except PresentationException as msg:
+                self.log_warning(f"Can not present diagnostic: {msg}")
+        return presentation_list
+
+    def get_template(self, context, template):
+        """get Jinja2 template file"""
+        search_path = [".", "templates"]
+        if "_se_cmd_cwd" in context:
+            search_path.extend(
+                [
+                    context["_se_cmd_cwd"],
+                    Path(context["_se_cmd_cwd"]) / "templates",
+                ]
+            )
+        self.log_debug(f"Search path for template: {search_path}")
+
+        loader = jinja2.FileSystemLoader(search_path)
+        environment = jinja2.Environment(loader=loader)
+        for name, function in j2filters().items():
+            environment.filters[name] = function
+        return environment.get_template(template)
+
+    def get_project_and_issue(self, gl, issue_subject):
+        """Connect to Gitlab server, find and return correct issue and project based on user input"""
+
+        project_id = 1982
```

Comment:
Should this become a module constant (with some comment)? Such that it is a little easier to find the day it changes.

Reply:
Addressed in [329a135](https://github.com/uwefladrich/scriptengine-tasks-ecearth/pull/95/commits/329a13542523906562d7a1e6972ad8e1853b2c12)